File: 2/PHP/Memory.php
<?php

class Memory extends Hardware{
	// Atribut Private
	private $frequency;
	private $memorySize;
	private $supportsCuda;

	// Konstruktor
	function __construct() {
		$frequency = 0;
		$memorySize = 0;
		$supportsCuda = "";
	}

	// Set & Get

	// Frequency
	function setFrequency($frequency) {
		$this->frequency = $frequency;
	}
	function getFrequency() {
		return $this->frequency;
	}

	// Memory Size
	function setMemorySize($memorySize) {
		$this->memorySize = $memorySize;
	}
	function getMemorySize() {
		return $this->memorySize;
	}

	// Cuda
	function setSupportsCuda($supportsCuda) {
		$this->supportsCuda = ($supportsCuda == 'y') ? "YES" : "NO";
	}
	function getSupportsCuda() {
		return $this->supportsCuda;
	}
}

?>

File: 2/Python/Memory.py
from Hardware import Hardware

class Memory(Hardware):
	# Konstruktor
	def __init__(self):
		self.__frequency = 0
		self.__memorySize = 0
		self.__supportsCuda = True

	# Set & Get

	# Frequency
	def setFrequency(self, frequency):
		self.__frequency = frequency

	def getFrequency(self):
		return self.__frequency

	# Memory Size
	def setMemorySize(self, memorySize):
		self.__memorySize = memorySize

	def getMemorySize(self):
		return self.__memorySize

	# Supports Cuda
	def setSupportsCuda(self, supportsCuda):
		self.__supportsCuda = True if (supportsCuda == 'y') else False

	def getSupportsCuda(self):
		return self.__supportsCuda